src/Agent.Listener/JobDispatcher.cs

```diff
@@ -265,6 +265,7 @@ public bool Cancel(JobCancelMessage jobCancelMessage)
 
                 if (jobRequestCancellationToken.IsCancellationRequested)
                 {
+                    notification.JobFinished(message.JobId);
```

Comment:
The others are on the line after `CompleteJobRequestAsync`, does it matter? `CompleteJobRequestAsync` is a private method, would it be better to move the call into there?